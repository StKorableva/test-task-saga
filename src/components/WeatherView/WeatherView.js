import React, { Component } from "react";
import Spinner from "../Spinner/Spinner";
import "./WeatherView.scss";

class WeatherView extends Component {
  componentDidMount() {
    this.props.getWeather();
  }

  componentDidUpdate(prevProps) {
    if (prevProps.city !== this.props.city) {
      this.props.getWeather();
    }
  }

  transformToC = temp => {
    return (temp - 273.15).toFixed(1);
  };

  render() {
    const { city, isFetching, weather } = this.props;
    if (isFetching) {
      return <Spinner />;
    } else {
      return (
        <div className="main">
          <h1 className="main__logo">{city}</h1>
          <p className="main__temp">
            Temperature: {this.transformToC(weather.temp)} â„ƒ
          </p>
        </div>
      );
    }
  }
}

export default WeatherView;
