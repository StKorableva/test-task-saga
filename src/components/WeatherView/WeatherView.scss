.main {
  color: #000;
  font-family: Arial;
  font-weight: 600;
  margin-bottom: 50px;
  text-transform: uppercase;
  z-index: 1;

  &__logo {
    font-size: 30px;
  }

  &__temp {
    font-size: 20px;
  }
}
