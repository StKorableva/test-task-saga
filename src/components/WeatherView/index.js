import { connect } from "react-redux";
import { getWeather } from "../../store/actions/weatherActions";
import WeatherView from "./WeatherView";
import selector from "../../selectors/weatherSelector";

const mapStateToProps = state => ({
  weather: selector(state),
  isFetching: state.isFetching,
  error: state.error
});

const mapDispatchToProps = (dispatch, ownProps) => ({
  getWeather() {
    dispatch(getWeather(ownProps.city));
  }
});
export default connect(mapStateToProps, mapDispatchToProps)(WeatherView);
