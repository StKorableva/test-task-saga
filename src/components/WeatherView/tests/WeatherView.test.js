import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import WeatherView from '../WeatherView';

const props = {
    getWeather: jest.fn(),
    weather: {
        temp: 280.5
    },
    isFetching: false,
    city: 'saint petersburg'
}

const weatherview = shallow(<WeatherView {...props} />);


describe('<WeatherView />', () => {
	it('should match snapshot', () => {
		expect(toJson(weatherview)).toMatchSnapshot();
    });
    
    it('should render Spinner when data is fetching', () => {
        weatherview.setProps({isFetching: true});
        expect(toJson(weatherview)).toMatchSnapshot();
    })

    it('update data when change city', () => {
        weatherview.setProps({city: 'moscow'});
        expect(weatherview.instance().props.getWeather).toHaveBeenCalledTimes(2);
    })
});
