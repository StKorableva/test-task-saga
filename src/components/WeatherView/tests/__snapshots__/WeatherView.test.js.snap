// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherView /> should match snapshot 1`] = `
<div
  className="main"
>
  <h1
    className="main__logo"
  >
    saint petersburg
  </h1>
  <p
    className="main__temp"
  >
    Temperature: 
    280.5
     
  </p>
</div>
`;

exports[`<WeatherView /> should render Spinner when data is loading 1`] = `<Spinner />`;
